### ISSUE TITLE
3rd Party Server Support

### ISSUE THREAD

**Issue by headap** [revoltchat/desktop#53]
The client should be able to log in to self-hosted servers besides the flagship revolt.app server.

I suggest a third field upon login (besides email and password) which will by default contain the flagship server url but can be easily changed prior to logging in

**Comment by Rexogamer** [comment]
We've discussed this internally and right now we don't currently have plans to add support for third-party instances. I know this is disappointing, but if we have any updates on this (or we decide on any paths involving third-party support) we'll make sure to share them.

**Comment by bitfl0wer** [comment]
Are there any updates on this? It would be really great to be able to comfortably use self-hosted instances without having to manipulate config files.

**Comment by aqua-pro** [comment]
Greetings, dear friends

Is there any update to this issue? I'd like to selfhost my own server instance and use the desktop client. I believe many people who are privacy-minded like me would want to do so, too, it's how I found out about Revolt in first place

If you are not going to add support, would you mind showing me on what files the connection string is? This way I'd distribute the binary files myself to my friends

Thank you

**Comment by Rexogamer** [comment]
The URLs can be found in [this function](https://github.com/revoltchat/desktop/blob/eb7f7b894eb4d3340961c0df2ebe8f35571a3244/src/lib/config.ts#L48) - replacing these with your own instance's Revite links should be sufficient

**Comment by bitfl0wer** [comment]
I am sorry to say it like this, but having to dig into config files and changing variables is absolutely not end-user-friendly. It would be great if you could choose the instance you'd like to connect to on login, via UI

**Comment by insertish** [comment]
> I am sorry to say it like this, but having to dig into config files and changing variables is absolutely not end-user-friendly. It would be great if you could choose the instance you'd like to connect to on login, via UI

Two things:
- if it were a relatively hidden option, (see BitWarden app / extension login), then I would potentially consider this but I don't personally have the time to work on such a feature, if someone wants to contribute this in then they can create a PR
- regarding why it should appear as such: my focus is on revolt.chat, I don't want to confuse users when they're signing up about what instance they're meant to use

There may be other security considerations here too.

**Comment by GanimanSwift** [comment]
This seems like it would be a relatively easy to add feature. Just dont hard-code the clients to connect to revolt.chat and make them variables, and put some option on the log in menu to enter a custom server URL and default it to revolt.chat. Its amazing that this is an open source project, but neglecting to add this type of feature goes against the spirit of open source. If you are trying to lock people into using Revolt on revolt.chat, then why even publish self-hosted methods.

**Comment by ImVexed** [comment]
Bump on this. Seems very strange to let people self-host the whole backend and access the same UI in the browser - yet gate keep setting the login url in the electron shim.

**Comment by aqua-pro** [comment]
Bump on this. Super easy change. privacy friendly, decentralized, giving people choice. All of these things are good things. Don't limit people's choice on which server to connect with your released binaries.

Just open it to the world, you've already opened the source code, why not add this option?

Why even have a docker backend if you need to modify and compile the binaries for anyone to connect to?

More people might use Revolt after this change. I would be one of them. I wanna use Revolt, but on my own terms.

Everything is ready except the hardcoded connection string. Just add an optional setting to change the server connection. Some people will appreciate it a lot.

**Comment by insertish** [comment]
Just re-opening the issue, anyone is free to tackle this.

**Comment by codevski** [comment]
> We've discussed this internally and right now we don't currently have plans to add support for third-party instances. I know this is disappointing, but if we have any updates on this (or we decide on any paths involving third-party support) we'll make sure to share them.

Mmm, It's either Discord holds data or Revolt holds data, who ever caves in first when lawyers request data...

At login most self-host privacy focused projects usually defaults to an official server with an option to connect to an alternative instance:

![Image](https://github.com/user-attachments/assets/7866a87e-cb68-44f0-af52-d659e261160a)

Notice this direction with the mobile apps as well. Hopefully project direction changes 🤷

**Comment by sharkaccino** [comment]
@codevski just fyi, you're replying to an over 3 year old comment. pretty sure the general consensus on this subject has changed a lot since then.

**Comment by aqua-pro** [comment]
3 years old, and it's still relevant as ever. This is the one change that stops me from fully using Revolt.

**Comment by Voroxpete** [comment]
> 3 years old, and it's still relevant as ever. This is the one change that stops me from fully using Revolt.

Same here.  I've got an entire Discord server looking to move to a different platform before the enshittification gets too bad, but this will be a deal-breaker.  I'm happy to maintain the back-end, but I'm not going to supply everyone with modified desktop clients or force them all to use the web client only.

**Comment by GanimanSwift** [comment]
> > 3 years old, and it's still relevant as ever. This is the one change that stops me from fully using Revolt.
> 
> Same here. I've got an entire Discord server looking to move to a different platform before the enshittification gets too bad, but this will be a deal-breaker. I'm happy to maintain the back-end, but I'm not going to supply everyone with modified desktop clients or force them all to use the web client only.

Its not the perfect solution, but see my [pull request](https://github.com/revoltchat/desktop/pull/164) that added this functionality through a command line argument. Use `server` and provide a url and it should connect to a self hosted instance. This might help you work around it.

**Comment by zierf** [comment]
I agree with many of the previous comments. Back in the Revolt days, I painstakingly tried to find a suitable setup for Docker containers to see if it could be a viable alternative to TeamSpeak (and Discord). At that time, there was no official `compose.yml` file and you had to use various snippets and projects, even replace the voice server with a beta version.

After I had everything basically working and the web UI was functioning as intended, I wanted to use the desktop client to log into the self-hosted instance. This was ultimately the dealbreaker then.

Compared to simply launching TeamSpeak or Discord, it's much more inconvenient to always have to open the frontend in the browser and then check for messages there every time. Aside from the easy and quick access, it is extremely useful to have a tray icon that can show you at a glance whether you are currently muted and whether the microphone is being triggered correctly (in case you forget the hardware mute).

A solution similar to the Matrix client Element, where the default address could be changed to a self-hosted instance, would probably be the most sensible. Now is actually a good time to do it. This could attract Discord users switching to the Stoat community and it might still be early enough before TeamSpeak6 gains too much traction (will be more Discord-like).

Not everyone should start their own fork, this leads to unnecessary fragmentation. Aside from the cumbersome and unnecessary additional effort, it would probably not be possible for every potential user to implement this themselves, which leads to further friction and ultimately impairs the project's adoption.
